File: PolygonEditorObjC/PolygonEditorObjC/PolygonOpen.cpp
#include <GLUT/glut.h>
#include <iostream>
#include "PolygonOpen.h"

using namespace std;

PolygonOpen::PolygonOpen() {
}

PolygonOpen::~PolygonOpen() {
}

void PolygonOpen::init() {
	ObjectGraph::init();
}

void PolygonOpen::draw() {
	ObjectGraph::draw();

	glBegin(GL_LINE_STRIP);
		size_t totalPoints = points.size();
		for (int i = 0; i < totalPoints; i++) {
			Point* point = points.at(i);
			glVertex2f(point->getX(), point->getY());
		}
	glEnd();
}

File: PolygonEditorObjC/PolygonEditorObjC/PolygonOpen.m
//
//  PolygonOpen.m
//  PolygonEditorObjC
//
//  Created by Silvio Fragnani on 17/10/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "PolygonOpen.h"


@implementation PolygonOpen

- (id)init
{
    self = [super init];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (void)dealloc
{
    [super dealloc];
}

- (void) start
{
	[super start];
}

- (void) draw
{
	[super draw];
	
	glBegin(GL_LINE_STRIP);

	for (int i = 0; i < [points count]; i++) {
		Pointer * point = [points objectAtIndex: i];
		glVertex2f([point x], [point y]);
	}
	glEnd();

}


@end
